app/views/building_map/index.html.erb

```diff
@@ -1,11 +1,131 @@
-<%=
-  map(
-    center: {
-        # HPI coordinates if no center is set
-        latlng: !@target.nil? ? @target.split(",") : [52.39339, 13.13208],
-        zoom: 18,
-    },
-    polygons: BuildingMapHelper.leaflet_polygons(),
-    polylines: BuildingMapHelper.leaflet_polylines(@route),
-    markers: BuildingMapHelper.leaflet_markers(@route, @target))
-%>
+<div class="offcanvas offcanvas-end" tabindex="-1" id="offcanvas" data-bs-keyboard="false" data-bs-backdrop="false">
+  <div class="offcanvas-header">
+    <h5 class="offcanvas-title d-none d-sm-block" id="offcanvas">Navigation</h5>
+    <button type="button" class="btn-close text-reset" data-bs-dismiss="offcanvas" aria-label="Close"></button>
+  </div>
+  <div class="offcanvas-body px-0">
+    <div class="accordion mb-2 accordion-flush" id="routingInput">
+      <div class="accordion-item">
+        <h2 class="accordion-header" id="headingOne">
+          <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse"
+                  data-bs-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
+            <i class="fas fa-directions directions-icon"></i>
+          </button>
+        </h2>
+        <div id="collapseOne" class="accordion-collapse collapse show" aria-labelledby="headingOne"
+             data-bs-parent="#routingInput">
+          <div class="accordion-body">
+            <form class="row gy-2 gx-3 align-items-center" autocomplete="off" id="navigation_form">
+              <div class="col-auto">
+                <div class="input-group">
+                  <div class="input-group-text"><i class="fas fa-map-marker-alt route-start-icon"></i></div>
+                  <input name='start' type="text" class="form-control" list="startOptions" id="start_input"
+                         placeholder="From" onchange="request_location()">
```

Comment:
I tried implementing your suggestion, but it didn't work out that well, because awaiting async functions (like `getCurrentPosition`) in event listeners is apparently a very complicated thing to do. I'm leaving the onchange handler here for now as a workaround (there will always be sufficient time between triggering the `onchange` and the `onsubmit` handler, so we should have the correct location when executing the `onsubmit` handler).

Reply:
Okay then :)